src/parsers/manifest/metaplaylist/index.ts

```diff
@@ -0,0 +1,172 @@
+      manifest: IParsedManifest;
+      url: string;
+      transport: "dash"|"smooth";
+      startTime: number;
+      endTime: number;
+      textTracks: Array<{
+        url: string;
+        language: string;
+        mimeType: string;
+      }>;
+  }>,
+  attributes: {
+    timeShiftBufferDepth: number;
+  },
+  baseURL: string
+): IParsedManifest {
+
+  const parsedPeriods = contents
+    .map((content) => content.manifest.periods[0]);
+
+  // Build manifest root attributes
+  const presentationLiveGap =
+  (contents.map(content => content.manifest.presentationLiveGap)
+    .reduce((acc, val) =>
+      Math.min(acc || DEFAULT_LIVE_GAP, val || DEFAULT_LIVE_GAP), DEFAULT_LIVE_GAP
+    )) || DEFAULT_LIVE_GAP;
+
+  const suggestedPresentationDelay =
+    (contents.map(content => content.manifest.suggestedPresentationDelay)
+      .reduce((acc, val) => Math.min(acc || 10, val || 10), 10)) || 10;
+
+  const maxSegmentDuration =
+    contents.map(content => content.manifest.maxSegmentDuration)
+      .reduce((acc, val) => Math.min((acc || 0), (val || 0)), 0);
+
+  const minBufferTime =
+    contents.map(content => content.manifest.minBufferTime)
+      .reduce((acc, val) => Math.min((acc || 0), (val || 0)), 0);
+
+  // Build new period array
+  const newPeriods: IParsedPeriod[] = [];
+  const contentEnding = contents[contents.length - 1].endTime;
+
+  for (let j = 0; j < parsedPeriods.length; j++) {
+    const baseManifest = new Manifest(contents[j].manifest as IManifestArguments);
+    const parsedPeriod = parsedPeriods[j];
+    parsedPeriod.start = contents[j].startTime;
+    parsedPeriod.end = parsedPeriod.start + (parsedPeriod.duration || 0);
+    const textTracks = contents[j].textTracks;
+    if (textTracks && textTracks.length > 0) {
+      textTracks.forEach((track) => {
+        const textAdaptation = {
+          id: "gen-text-ada-" + generateNewId(),
+          representations: [{
+            mimeType: track.mimeType,
+            bitrate: 0,
+            index: new StaticRepresentationIndex({
+              media: track.url,
+              startTime: parsedPeriod.start,
+              endTime: parsedPeriod.end || Number.MAX_VALUE,
+            }),
+            id: "gen-text-rep-" + generateNewId(),
+          }],
+          type: "text",
+          normalizedLanguage: track.language,
+        };
+        parsedPeriod.adaptations.push(textAdaptation);
+      });
+    }
+
+    const adaptations = parsedPeriod.adaptations;
+    adaptations.forEach((adaptation) => {
+      const representations = adaptation.representations;
+      representations.forEach((representation) => {
+        const index = representation.index;
+
+        // Store base contents info
+        const baseAdaptation = ["audio", "video", "image", "text"]
+          .reduce((acc: Adaptation[], type) => {
+            const _adaptation = baseManifest.adaptations[type as AdaptationType];
+            if (_adaptation) {
+              acc.concat(_adaptation);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: The return value of "concat" must be used. ([typescript:S2201](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS2201))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWLjl6xr-pl6AHs2a-uO&open=AWLjl6xr-pl6AHs2a-uO)

[ISSUE_KEY]: AWLjl6xr-pl6AHs2a-uO